gmime/gmime-parser.c

```diff
@@ -975,6 +962,18 @@ has_content_headers (GPtrArray *headers)
 	return FALSE;
 }
 
+#define warn_bad_header G_STMT_START { 												\
+	if (can_warn) { 													\
+		gchar *eol; 													\
+		gchar *bad_header; 												\
+		 	 	 	 	 	 	 	 	 	 	 				\
+		for (eol = inptr; eol < inend && eol[0] != '\r' && eol[0] != '\n'; eol++); 					\
+		bad_header = g_strndup (start, eol - start); 									\
+		_g_mime_parser_options_warn (options, priv->header_offset, GMIME_CRIT_INVALID_HEADER_NAME, bad_header);		\
+		g_free (bad_header); 												\
+	} 															\
+} G_STMT_END
+
```

Comment:
I'd probably prefer that this be a function rather than a macro.

Reply:
I agree that the macro is ugly.  As a function would require 5 (or 6, if we pass can_warn) parameters, it seemed to be easier to me, though.  I will fix that…